libs/go/sia/host/provider/provider.go

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright The Athenz Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package provider
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"net"
+	"net/url"
+
+	"github.com/AthenZ/athenz/libs/go/sia/host/ip"
+	"github.com/AthenZ/athenz/libs/go/sia/host/signature"
+)
+
+// Provider is the interface which wraps various Providers known to ZTS
+// It has methods for providing attestationdata depending on provider type
+// and generating sub-parts of DN to be including in the CSR and San DNS and URI entries
+type Provider interface {
+	// PrepareKey creates/setup up a private key for use with the provider
+	PrepareKey(string) (*rsa.PrivateKey, error)
```

Comment:
isn't there an interface that we can use here instead of rsa? We should be able to support both rsa and ec keys.

Reply:
switching to crypto.Privatekey